File: ui/lib/core/models/banner.dart
import '../utils/url_util.dart';

class BannerModel {
  final int id;
  final String title;
  final String? subtitle;
  final String? description;
  final String imageUrl;
  final int? discountPercentage;
  final String? buttonText;
  final String? actionType;
  final String? actionId;
  final String? actionUrl;
  final int displayOrder;

  BannerModel({
    required this.id,
    required this.title,
    this.subtitle,
    this.description,
    required this.imageUrl,
    this.discountPercentage,
    this.buttonText,
    this.actionType,
    this.actionId,
    this.actionUrl,
    required this.displayOrder,
  });

  factory BannerModel.fromJson(Map<String, dynamic> json) {
    return BannerModel(
      id: json['id'] is int ? json['id'] : int.tryParse(json['id'].toString()) ?? 0,
      title: json['title'] ?? '',
      subtitle: json['subtitle'],
      description: json['description'],
      imageUrl: UrlUtil.formatImageUrl(json['image_url'] ?? ''),
      discountPercentage: json['discount_percentage'] is int
          ? json['discount_percentage']
          : int.tryParse(json['discount_percentage']?.toString() ?? ''),
      buttonText: json['button_text'],
      actionType: json['action_type'],
      actionId: json['action_id']?.toString(),
      actionUrl: json['action_url'],
      displayOrder: json['display_order'] is int
          ? json['display_order']
          : int.tryParse(json['display_order']?.toString() ?? '0') ?? 0,
    );
  }
}

File: svelte_demo/app/Models/Banner.php
<?php

namespace App\Models;

use App\Traits\HasOptimizedImages;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Banner extends Model
{
    /** @use HasFactory<\Database\Factories\BannerFactory> */
    use HasFactory, HasOptimizedImages;

    protected $fillable = [
        'title',
        'subtitle',
        'description',
        'image_url',
        'discount_percentage',
        'button_text',
        'action_type',
        'action_id',
        'action_url',
        'display_order',
        'status',
        'start_date',
        'end_date',
    ];

    protected function casts(): array
    {
        return [
            'display_order' => 'integer',
            'start_date' => 'datetime',
            'end_date' => 'datetime',
        ];
    }
}
